src/mpi/comm/comm_split_type.c

```diff
@@ -366,27 +366,67 @@ static int node_split_gpu_device(MPIR_Comm * comm_ptr, int key,
 
 static const char *SHMEM_INFO_KEY = "shmem_topo";
 
-static int compare_info_args(const void *sendbuf, void *recvbuf, int count, MPIR_Comm * comm_ptr,
-                             int *info_args_global)
+static int compare_info_hint(const char *hintval, MPIR_Comm * comm_ptr, int *info_args_are_equal)
 {
+    int hintval_size = strlen(hintval);
+    int hintval_size_global;
+    char *hintval_global = NULL;
     int mpi_errno = MPI_SUCCESS;
     MPIR_Errflag_t errflag = MPIR_ERR_NONE;
-    int info_args_are_equal = 0;
+    int info_args_global = 0;
+    int info_args_equal = 0;
```

Comment:
Please use more descriptive variable names.

Reply:
I have renamed the variables and added a comment describing them